добавлены классы для поиска животных

SOProfile opened the pull request:
добавлены классы для поиска животных, метод создания возвращает мапу, добавлены тесты для новых методов

Commit:
добавлены тесты и классы исключений

Commit:
добавлены тесты и классы исключений

NosikovNeOleg commented on src/main/java/AnimalsRepositoryImpl.java:
```diff
@@ -0,0 +1,83 @@
+        for (Integer count = 0; count < animals.size(); count++) {
+            switch (animals.get(count).getClass().getName()) {
+                case ("Cat"):
+                    cats = cats + 1;
+                    break;
+                case ("Dog"):
+                    dogs = dogs + 1;
+                    break;
+                case ("Wolf"):
+                    wolfs = wolfs + 1;
+                    break;
+                case ("Lion"):
+                    lions = lions + 1;
```
Чуть лучше будет выглядеть += или ++ вместо такого присвоения

NosikovNeOleg commented on src/main/java/AnimalsRepositoryImpl.java:
```diff
@@ -0,0 +1,83 @@
+                    break;
+            }
+        }
+        if (cats != 0) {
+            animalsDuplicate.put("Cats", cats);
+        }
+        if (dogs != 0) {
+            animalsDuplicate.put("Dogs", dogs);
+        }
+        if (wolfs != 0) {
+            animalsDuplicate.put("Wolfs", wolfs);
+        }
+        if (lions != 0) {
+            animalsDuplicate.put("Lions", lions);
+        }
```
Кажется в логике ошибка, если будет всего один кот, то метод выведет Cat=1?

NosikovNeOleg replied:
А хотя в задании вроде считается валидным..

NosikovNeOleg replied:
Раз по заданию не надо именно дубли искать, то всё ок

SOProfile replied:
ага, там нужно вывести количество животных

NosikovNeOleg commented on src/test/java/AnimalRepositoryTest.java:
```diff
@@ -0,0 +1,137 @@
+    Lion lion1 = new Lion("Порода1", "Имя1", "Характер1", 333.33, LocalDate.of(2024,1, 1));
+    Lion lion2 = new Lion("Порода1", "Имя1", "Характер1", 333.33, LocalDate.of(2023,1, 1));
+
+    List<Animal> animalsTestList = new ArrayList<>();
```
Если используется один список для всех тестов, то корректно будет его очищать после каждого теста

NosikovNeOleg replied:
Или до

Commit:
добавлены методы для:
1. поиска среднего возраста животных
2. поиска животных, возраст которых больше 5 и цена выше средней
3. поиска 3 животных с самой низкой ценой, отсортированных в обратном алф. порядке
4. поправила метод для проверки года рождения животного на високосность

методы класса animalRepositoryImpl переведены на stream api

NosikovNeOleg commented on src/main/java/AnimalsRepositoryImpl.java:
```diff
@@ -0,0 +1,83 @@
+        for (Integer count = 0; count < animals.size(); count++) {
+            switch (animals.get(count).getClass().getName()) {
+                case ("Cat"):
+                    cats += cats;
```
Чуть не то я имел в виду) Так получается, что к нулю каждый раз добавляется ноль и переменная не меняется

SOProfile replied:
поправила вроде)

Commit:
добавлены методы для:
1. поиска среднего возраста животных
2. поиска животных, возраст которых больше 5 и цена выше средней
3. поиска 3 животных с самой низкой ценой, отсортированных в обратном алф. порядке
4. поправила метод для проверки года рождения животного на високосность

методы класса animalRepositoryImpl переведены на stream api
